hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/V2MessageUtil.java

```diff
@@ -167,6 +171,21 @@ public static String getMsgId(String v2Message) {
 	
 		return msgId;
 	}
+	
+
+	public static String getMsgCnrtlId(String v2Msg) {
+		String methodName = LoggingUtil.getMethodName();
+		String controlId = null;
+		if (StringUtils.isBlank(v2Msg)) {
+			logger.warn("{} - MessageControlId is blank", methodName);
+        } else {
```

Comment:
This is really personal preference but I prefer to have the positive /most likelycase first an if statement (unless you are doing a validation check and then returning right away). So in this case, if the vMsg is valid (not blank) you extract the value, otherwise warn. It just makes the logic easier to follow.